Prototyping/Kayla-Camera/webApp/kaylaCamRouter.js

```diff
@@ -49,7 +50,7 @@ router.get('/settings', (req, res) => {
 router.post('/settings', (req, res) => {
     let query = `
         INSERT INTO \`settings\` (config_name, settings)
-        VALUES ('${req.body.configName}', '${JSON.stringify(req.body)}')`
+        VALUES ('${req.body.configName}', '${mysql.escape(req.body)}')`
```

Comment:
doesn't `req.body.configName` also have to be escaped?

also, does this work? i thought `escape` was a method on a `connection` object?

Reply:
These all call the same thing - In order to avoid SQL Injection attacks, you should always escape any user provided data before using it inside a SQL query. You can do so using the mysql.escape(), connection.escape() or pool.escape() methods:  

Maybe I'm a little confused, wouldn't that mean everything we are doing needs to be calling this?   For example in hifiPing:
```    var query = `REPLACE INTO \`subscriptions\` (username, subscription)
        VALUES ('${body.username}', '${JSON.stringify(body.subscription)}')`;```

Reply:
nice, ok, cool.

as for your second paragraph: yes, any user-submitted data that is going to be part of a MySQL query should be escaped. so yes, we (i, in this example) should really go back and update old code to prevent SQL Injection. it's a very low-priority issue right now, though.

Reply:
I gotcha though, definitely more important for user-submitted values.  Will fix this.